cubical/machines/slope_machine.py

```diff
@@ -255,18 +257,26 @@ def implement_update(self, jhr, jhjinv):
 
         self.slope.construct_gains(self.slope_params, self.gains, self.chunk_ts, self.chunk_fs, self.t_int, self.f_int)
 
+
     def restrict_solution(self, slope_params):
         """
         Restricts the solution by invoking the inherited restrict_solution method and applying
         any machine specific restrictions.
         """
 
-        # ParameterisedGains.restrict_solution(self)
+        #ParameterisedGains.restrict_solution(self)
         
+        # These need to be set here as we do not call the interval restrict solutions - 
+        # out solutions are the parameters and not the gains themselves.
+        self._gh_update = self._ghinv_update = True
+
         if self.ref_ant is not None:
             # complicated slice :) we take the 0,0 phase offset of the reference antenna,
             # and subtract that from the phases of all other antennas and elements
-            slope_params -= slope_params[:,:,:,self.ref_ant,:,0,0][:,:,:,np.newaxis,:,np.newaxis,np.newaxis]
+            ref_slice = slice(self.ref_ant, self.ref_ant+1, 1)
+            ref_params = slope_params[:,:,:,ref_slice,:,(0,1),(0,1)]
+            slope_params[:,:,:,:,:,(0,1),(0,1)] -= ref_params
```

Comment:
I'm not sure this is correct @JSKenyon -- I think only the 0,0 slope must be subtracted. In the presence of any polarized power, the RL or XY phase-zero difference is constrained by the off-diagonal data.

This does give me a most happy thought -- the PZD solution really belongs here and not in complex-2x2. It should be as easy as solving for two f-slopes.

Reply:
Agreed @o-smirnov. Just pushed a minor fix to this branch to implement the correct reference behaviour.